Source/WebCore/platform/MIMETypeRegistry.cpp

```diff
@@ -561,8 +562,10 @@ bool MIMETypeRegistry::isSupportedJSONMIMEType(const String& mimeType)
     // When detecting +json ensure there is a non-empty type / subtype preceeding the suffix.
     if (mimeType.endsWithIgnoringASCIICase("+json"_s) && mimeType.length() >= 8) {
         size_t slashPosition = mimeType.find('/');
-        if (slashPosition != notFound && slashPosition > 0 && slashPosition <= mimeType.length() - 6)
-            return true;
+        if (slashPosition != notFound && slashPosition > 0 && slashPosition <= mimeType.length() - 6) {
+            if (isValidHTTPToken(mimeType.substring(slashPosition + 1)))
```

Comment:
We could verify until `mimeType.length() - 5` since we know it ends with `+json` (similar to `MIMETypeRegistry::isXMLMIMEType`).

We can also clean up `MIMETypeRegistry::isXMLMIMEType` since it's defined the same way as JSON (in this change or another)

Reply:
Also according to the spec, the type (not just the subtype) needs to be a valid HTTP token as well, but unfortunately there is no test coverage for that

Reply:
Sure, we can just check up to `mimeType.length() - 5`.